Source/GSStoryboardTransform.m

```diff
@@ -0,0 +1,878 @@
+// Private to this class...
+- (void) _setKind: (NSString *)k;
+- (void) _setRelationship: (NSString *)r;
+- (NSString *) _kind;
+- (NSString *) _relationship;
+@end
+
+// this needs to be set on segues
+@implementation NSStoryboardSegue (__StoryboardPrivate__)
+- (void) _setKind: (NSString *)k
+{
+  ASSIGN(_kind, k);
+}
+
+- (void) _setRelationship: (NSString *)r
+{
+  ASSIGN(_relationship, r);
+}
+
+- (NSString *) _kind
+{
+  return _kind;
+}
+
+- (NSString *) _relationship
+{
+  return _relationship;
+}
+@end
+
+@implementation NSStoryboardSeguePerformAction
+- (id) target
+{
+  return _target;
+}
+
+- (void) setTarget: (id)target
+{
+  ASSIGN(_target, target);
+}
+
+- (SEL) action
+{
+  return _action;
+}
+
+- (void) setAction: (SEL)action
+{
+  _action = action;
+}
+
+- (NSString *) selector
+{
+  return NSStringFromSelector(_action);
+}
+
+- (void) setSelector: (NSString *)s
+{
+  _action = NSSelectorFromString(s);
+}
+
+- (id) sender
+{
+  return _sender;
+}
+
+- (void) setSender: (id)sender
+{
+  ASSIGN(_sender, sender);
+}
+
+- (NSString *) identifier
+{
+  return _identifier;
+}
+
+- (void) setIdentifier: (NSString *)identifier
+{
+  ASSIGN(_identifier, identifier);
+}
+
+- (NSString *) kind
+{
+  return _kind;
+}
+
+- (void) setKind: (NSString *)kind
+{
+  ASSIGN(_kind, kind);
+}
+
+- (NSStoryboard *) storyboard
+{
+  return _storyboard;
+}
+
+- (void) setStoryboard: (NSStoryboard *)storyboard
+{
+  ASSIGN(_storyboard, storyboard);
+}
+
+- (NSStoryboardSegue *) storyboardSegue
+{
+  return _storyboardSegue;
+}
+
+- (void) setStoryboardSegue: (NSStoryboardSegue *)ss
+{
+  ASSIGN(_storyboardSegue, ss);
+}
+
+- (void) dealloc
+{
+  RELEASE(_storyboard);
+  RELEASE(_kind);
+  RELEASE(_identifier);
+  RELEASE(_sender);
+  RELEASE(_storyboardSegue);
+  [super dealloc];
+}
+
+- (IBAction) doAction: (id)sender
+{
+  BOOL should = YES;
+  BOOL responds = [_sender respondsToSelector: @selector(shouldPerformSegueWithIdentifier:sender:)];
+
+  if (responds)
+    {
+      should = [_sender shouldPerformSegueWithIdentifier: _identifier
+                                                  sender: _sender];
+    }
+
+  if (should)
+    {
+      id destCon = nil;
+      if ([[_storyboardSegue destinationController] isKindOfClass: [NSViewController class]] ||
+          [[_storyboardSegue destinationController] isKindOfClass: [NSWindowController class]])
+        {
+          destCon = [_storyboardSegue destinationController];
+        }
+      else
+        {
+          NSString *destId = [_storyboardSegue destinationController];
+          destCon = [_storyboard instantiateControllerWithIdentifier: destId]; 
+        }
+      [_storyboardSegue _setSourceController: _sender]; 
+      [_storyboardSegue _setDestinationController: destCon];  // replace with actual controller...
+      
+      if (_sender != nil)
+        {
+          [_sender performSegueWithIdentifier: _identifier
+                                       sender: _sender];
+        }
```

Comment:
The else case is missing here

Reply:
Why? Should remains YES.

Reply:
I added an explanation as to why there is no else.

Reply:
I am not able to see that explanation and previously you had a [_storyboardSegue perform]; here. Was that code wrong?

Reply:
As nil does not respond to any selector you could now remove that check. But more importantly you are still missing the [_storyboardSegue perform] call in the else case of this if statement.
Sometimes I have the impression that I am not expressing my comments clear enough. On the other hand I don't know what else to write to make it more explicit.

Reply:
Do you actually read my comments? Sometimes I doubt it.

Reply:
"an else case is missing" that's really explicit, Fred. </sarcasm>

Reply:
OK, let's try again:

Could you please read the code 

else
{
 [_storyboardSegue perform]; 
}

here?

Reply:
Given that the code below that is performing that function fine is why I was confused.  If you look at the code below it is resolving the destination controller which *still* needs to be looked up even in the special case.  So I was not seeing your point.    The additional check that is done on _sender is to make certain that the performSegueWithIdentifier: method is not called on an object that doesn't have it.  The else fits best there.   Sorry for the misunderstanding.